### PR TITLE
ci: add logictest in cluster mode

### PR DESCRIPTION
**Pull request by ZeaLoVe**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

The logic test will run in cluster mode. Only run base suites for now.
Original stateless support using *_cluster.result to read different expected results on different modes. But it does not support in logictest. Add a dir named mode under suites, put cluster and standalone in it. Add command in the logictest actions after base suites are finished.

`python3 main.py --suite suites/mode --run-dir cluster`

Fixes #issue

### CODE REVIEW

**Comment by zhang2014** [comment]
@mergify update

**Comment by Xuanwo** [comment]
@mergify update

File: .github/actions/test_sqllogic_cluster_linux/action.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        profile: ${{ inputs.profile }}
+        sha: ${{ github.sha }}
+        target: ${{ inputs.target }}
+
+    - name: Test setup
+      shell: bash
+      run: |
+        bash ./scripts/setup/dev_setup.sh -yd
+
+    - name: Run sqllogic Tests with Cluster mode
```
Comment (Xuanwo) [inline .github/actions/test_sqllogic_cluster_linux/action.yml:31]:
```suggestion
    - name: Run sqllogic Tests with Cluster mode
```

**Comment by Xuanwo** [comment]
Fixed the yamllint error to make @ZeaLoVe a nice day!

**Comment by ZeaLoVe** [comment]
cc @leiysky @zhang2014  do you agree with the way here to take place of the original stateless test with cluster mode? Any ideas?

**Comment by zhang2014** [comment]
> cc @leiysky @zhang2014 do you agree with the way here to take place of the original stateless test with cluster mode? Any ideas?

👍 Look good to me

**Comment by leiysky** [comment]
> cc @leiysky @zhang2014 do you agree with the way here to take place of the original stateless test with cluster mode? Any ideas?

LGTM, too.

File: tests/logictest/suites/base/15_query/cte.test [hunk 2]
```diff
@@ -103,7 +103,7 @@ WITH test1 AS (SELECT i + 1, j + 1 FROM test1) SELECT * FROM (SELECT * FROM test
 4 5
 
 statement query III
-SELECT * FROM (WITH t1 AS (SELECT to_int32(number) i FROM numbers(5)) SELECT * FROM t1) l INNER JOIN test1 r on l.i = r.i;
+SELECT * FROM (WITH t1 AS (SELECT to_int32(number) i FROM numbers(5)) SELECT * FROM t1) l INNER JOIN test1 r on l.i = r.i order by l.i, r.j;
```
Comment (xudong963) [inline tests/logictest/suites/base/15_query/cte.test:106]: LGTM, thank you @ZeaLoVe

**Comment by ZeaLoVe** [comment]
image

Cluster mode cost twice of time to finish base suites, need to optimize later, cc @zhang2014 , All test pass, we can merge this. logictest cluster mode will be optional for one or two weeks.(mergiy config file will be modified later)

**Review by zhang2014 — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

**Review by leiysky — APPROVED** [review]

**Comment by xudong963** [comment]
@mergify update

**Pull request merged**